src/NzbDrone.Core/Notifications/CustomScript/CustomScriptService.cs

```diff
@@ -0,0 +1,82 @@
+        {
+            var environmentVariables = new StringDictionary();
+
+            environmentVariables.Add("Sonarr.EventType", "Download");
+            environmentVariables.Add("Sonarr.Series.Id", series.Id.ToString());
+            environmentVariables.Add("Sonarr.Series.Title", series.Title);
+            environmentVariables.Add("Sonarr.Series.Path", series.Path);
+            environmentVariables.Add("Sonarr.Series.TvdbId", series.TvdbId.ToString());
+            environmentVariables.Add("Sonarr.EpisodeFile.Id", episodeFile.Id.ToString());
+            environmentVariables.Add("Sonarr.EpisodeFile.RelativePath", episodeFile.RelativePath);
+            environmentVariables.Add("Sonarr.EpisodeFile.Path", Path.Combine(series.Path, episodeFile.RelativePath));
+            environmentVariables.Add("Sonarr.EpisodeFile.SeasonNumber", episodeFile.SeasonNumber.ToString());
+            environmentVariables.Add("Sonarr.EpisodeFile.EpisodeNumbers", String.Join(",", episodeFile.Episodes.Value.Select(e => e.EpisodeNumber)));
+            environmentVariables.Add("Sonarr.EpisodeFile.EpisodeAirDates", String.Join(",", episodeFile.Episodes.Value.Select(e => e.AirDate)));
+            environmentVariables.Add("Sonarr.EpisodeFile.EpisodeAirDatesUtc", String.Join(",", episodeFile.Episodes.Value.Select(e => e.AirDateUtc)));
+            environmentVariables.Add("Sonarr.EpisodeFile.Quality", episodeFile.Quality.Quality.Name);
+            environmentVariables.Add("Sonarr.EpisodeFile.QualityVersion", episodeFile.Quality.Revision.Version.ToString());
+            environmentVariables.Add("Sonarr.EpisodeFile.ReleaseGroup", episodeFile.ReleaseGroup);
+            environmentVariables.Add("Sonarr.EpisodeFile.SceneName", episodeFile.SceneName);
+            
+            ExecuteScript(environmentVariables, settings);
+        }
+
+        public void AfterRename(Series series, CustomScriptSettings settings)
+        {
+            var environmentVariables = new StringDictionary();
+
+            environmentVariables.Add("Sonarr.EventType", "Rename");
+            environmentVariables.Add("Sonarr.Series.Id", series.Id.ToString());
+            environmentVariables.Add("Sonarr.Series.Title", series.Title);
+            environmentVariables.Add("Sonarr.Series.Path", series.Path);
+            environmentVariables.Add("Sonarr.Series.TvdbId", series.TvdbId.ToString());
+
+            ExecuteScript(environmentVariables, settings);
+        }
+
+        public ValidationFailure Test(CustomScriptSettings settings)
+        {
+            return null;
+        }
+
+        private void ExecuteScript(StringDictionary environmentVariables, CustomScriptSettings settings)
+        {
+            _logger.Debug("Executing external script: {0}", settings.Path);
+
+            var process = _processProvider.StartAndCapture(settings.Path, settings.Arguments, environmentVariables);
```

Comment:
we should store the output somewhere. otherwise it would be really annoying for people to debug it.